src/components/profileCard.tsx

```diff
@@ -0,0 +1,49 @@
+export interface User {
+  user_id: string;
+  first_name?: string;
+  last_name?: string;
+  location?: string;
+  birthday?: string | null;
+  bio?: string;
+  profile_picture?: string | null;
+  favourite_book?: string | null;
+}
+
+export default function UserProfileCard({
+  username,
+  user,
+}: {
+  username: string;
+  user: User;
+}) {
+  // const firstName = localStorage.getItem("firstName") || "";
+  // const lastName = localStorage.getItem("lastName") || "";
+  // const location = localStorage.getItem("location") || "";
+  // const bio = localStorage.getItem("about") || "";
+
+  const bio = user.bio;
+  const firstName = user.first_name;
+  const lastName = user.last_name;
+  const location = user.location;
+
+  const fullname = `${firstName} ${lastName}`.trim();
```

Comment:
```suggestion
  const fullname = [firstName, LastName].filter(Boolean).join(" ").trim();
```
This way you filter out undefined values instead of rendering `undefined` if one of the names is undefined